### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by liliamyhaliuk**
Adding exercises: 1.8 (Exercise 4), 2.12 (Exercise 2, 3).

### CODE REVIEW

File: chapter1/1.9_exercise_2.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# 2. How many miles are there in 10 kilometers? Hint: there are 1.61 kilometers in a mile.
+
+full_distance = 10 / 1.61
+
+# 3. If you run a 10 kilometer race in 42 minutes 42 seconds, 
+#    what is your average pace (time per mile in minutes and seconds)? 
+#    What is your average speed in miles per hour?
+
+#average pace (time per mile in minutes and seconds)
+avg_pace_sec = total_seconds / full_distance
```
Comment (auraz) [inline chapter1/1.9_exercise_2.py:18]: this should be printed too

File: chapter2/2.10_exercise_2.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+#Exercise 2
+#using the Python as a calculator
+
+import math
+import datetime
+
+# 1. The volume of a sphere with radius r is 4/3 π r3. What is the volume of a sphere with radius 5? 
+radius = 5
+print('The volume of a sphere: ', (4 / 3 * math.pi * radius**3))
```
Comment (auraz) [inline chapter2/2.10_exercise_2.py:12]: JFYI that in python2, it will be the wrong answer.

File: chapter1/1.9_exercise_2.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#Exercises of the book "Think python"
```
Comment (auraz) [inline chapter1/1.9_exercise_2.py:1]: The file should be named 1.9.2.py

File: chapter1/1.9_exercise_2.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#1.8 Exercises
+
+#Exercise 4
+# 1. How many seconds are there in 42 minutes 42 seconds?
```
Comment (auraz) [inline chapter1/1.9_exercise_2.py:5]: Very good, that you've included tasks! 👍🏻

File: chapter2/2.10_exercise_2.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+#2. Suppose the cover price of a book is $24.95, but bookstores get a 40% discount. 
+#   Shipping costs $3 for the first copy and 75 cents for each additional copy.
+#   What is the total wholesale cost for 60 copies?
+cover_price = 24.95
+discount = 0.4
+copies_amount = 60
+
+
+s_cost_first = 3 #shipping_cost for the first copy
```
Comment (auraz) [inline chapter2/2.10_exercise_2.py:23]: JDYI PEP8 requires 2 spaces before comment and 1 space after #.

File: chapter2/2.10_exercise_2.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+s_cost_add = 0.75 #shipping_cost for each additional copy
+
+#total wholesale cost
+sum = (cover_price*discount) * copies_amount + (s_cost_add * (copies_amount - 1)) + s_cost_first
+print('Total wholesale cost for 60 copies: ', sum)
```
Comment (auraz) [inline chapter2/2.10_exercise_2.py:28]: Answer is different, need to investigate

Comment (auraz) [reply chapter2/2.10_exercise_2.py:28]: Should be (1-discount).

Comment (auraz) [reply chapter2/2.10_exercise_2.py:28]: Also ensure precision is correct.

File: chapter2/2.10_exercise_2.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+#3. If I leave my house at 6:52 am and run 1 mile at an easy pace (8:15 per mile), 
+#   then 3 miles at tempo (7:12 per mile) and 1 mile at easy pace again, 
+#   what time do I get home for breakfast?
+
+#speed (minutes)
+easy_pace = '08:15'
+tempo = '07:12'
+
+#function convert time from str format('07:12', MM:SS) to amount of seconds
```
Comment (auraz) [inline chapter2/2.10_exercise_2.py:39]: Very good that you're wring docstring!
According to PEP8, they should be written as:
```
def time_to_seconds(time: str):
     """Function convert time from str format('07:12', MM:SS) to amount of seconds."""
      return (int(time[ : time.find(':')]) * 60) + (int(time[time.find(':') + 1 : ]))
```

File: chapter2/2.10_exercise_2.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+#start time of the run
+start_time = datetime.datetime.strptime('6:52','%H:%M')
+
+#calculate duration of the run in seconds (miles*time per mile)
+whole_time = (2 * easy_pace + 3 * tempo)
+
+#calculate time of the end of the run
+home_time = (start_time + datetime.timedelta(seconds = whole_time)).time()
+
+print("You will get home at: ", home_time)
```
Comment (auraz) [inline chapter2/2.10_exercise_2.py:56]: This is correct!

**Review by auraz — CHANGES_REQUESTED** [review]
Looking very good, but require some improvements.

File: chapter2/2.10_exercise_2.py (hunk 5 above)
Comment (auraz) [reply chapter2/2.10_exercise_2.py:23]: `s_cost_first = 3  # Shipping_cost for the first copy.

Comment (auraz) [reply chapter2/2.10_exercise_2.py:23]: `s_cost_first = 3  # Shipping_cost for the first copy.

File: chapter2/2.10_exercise_2.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
+#Exercises of the book "Think python"
```
Comment (auraz) [inline chapter2/2.10_exercise_2.py:1]:
```
"""
Exercises of the book "Think python"
2.10 Exercises
"""
```

**Review by auraz — CHANGES_REQUESTED** [review]

File: chapter1/1.9_exercise_2.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+#average pace (time per mile in minutes and seconds)
+avg_pace_sec = total_seconds / full_distance
+avg_pace_min = avg_pace_sec // 60 + (int(avg_pace_sec % 60) / 100)
+
+print(avg_pace_min)
+
+#average speed in miles per hour
+avg_speed = 3600 / avg_pace_sec
+
+print(avg_speed)
```
[the remark this answers is not in the document]
Comment (liliamyhaliuk) [reply chapter1/1.9_exercise_2.py:26]: Done.

File: chapter1/1.9_exercise_2.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+full_distance = 10 / 1.61
+
+# 3. If you run a 10 kilometer race in 42 minutes 42 seconds, 
+#    what is your average pace (time per mile in minutes and seconds)? 
+#    What is your average speed in miles per hour?
+
+#average pace (time per mile in minutes and seconds)
+avg_pace_sec = total_seconds / full_distance
+avg_pace_min = avg_pace_sec // 60 + (int(avg_pace_sec % 60) / 100)
```
[the remark this answers is not in the document]
Comment (liliamyhaliuk) [reply chapter1/1.9_exercise_2.py:19]: It's a mistake. We don't need to do this operation here. I deleted it.

File: chapter1/1.9_exercise_2.py (hunk 1 above)
Comment (liliamyhaliuk) [reply chapter1/1.9_exercise_2.py:18]: Done.

File: chapter2/2.10_exercise_2.py (hunk 5 above)
Comment (liliamyhaliuk) [reply chapter2/2.10_exercise_2.py:23]: I installed the plugin "black" as needed so it fixed these mistakes. Thank you!

File: chapter2/2.10_exercise_2.py (hunk 6 above)
Comment (liliamyhaliuk) [reply chapter2/2.10_exercise_2.py:28]: Found this mistake. The answer should be correct now.
Also, I took care of precision, thank you.

File: chapter2/2.10_exercise_2.py (hunk 7 above)
Comment (liliamyhaliuk) [reply chapter2/2.10_exercise_2.py:39]: I added docstring to the right place in this function. Thank you!

File: chapter1/1.9_exercise_2.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (auraz) [reply chapter1/1.9_exercise_2.py:19]: Thank you

File: chapter1/1.9_exercise_2.py (hunk 3 above)
Comment (auraz) [reply chapter1/1.9_exercise_2.py:1]: Thank you

File: chapter2/2.10_exercise_2.py (hunk 5 above)
Comment (auraz) [reply chapter2/2.10_exercise_2.py:23]: Thank you

File: chapter2/2.10_exercise_2.py (hunk 6 above)
Comment (auraz) [reply chapter2/2.10_exercise_2.py:28]: Thank you

File: chapter2/2.10_exercise_2.py (hunk 7 above)
Comment (auraz) [reply chapter2/2.10_exercise_2.py:39]: Thank you

**Pull request merged**